kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtCloudServiceImpl.java

```diff
@@ -34,34 +38,99 @@
 
     private static final long serialVersionUID = 2595835826149606703L;
 
-    private static final String COMPONENT_NAME = "component.name";
-    private static final String SERVICE_FACTORY_PID = "service.factoryPid";
+    private static final String KURA_SERVICE_PID = "kura.service.pid";
     private static final String KURA_UI_CSF_PID_DEFAULT = "kura.ui.csf.pid.default";
     private static final String KURA_UI_CSF_PID_REGEX = "kura.ui.csf.pid.regex";
 
+    private static final Comparator<GwtCloudConnectionEntry> CLOUD_CONNECTION_ENTRY_COMPARTOR = new Comparator<GwtCloudConnectionEntry>() {
+
+        @Override
+        public int compare(GwtCloudConnectionEntry o1, GwtCloudConnectionEntry o2) {
+            int rc;
+
+            rc = o1.getCloudFactoryPid().compareTo(o2.getCloudFactoryPid());
+            if (rc != 0) {
+                return rc;
+            }
+
+            rc = o1.getCloudServicePid().compareTo(o2.getCloudServicePid());
+            if (rc != 0) {
```

Comment:
True, it just keeps the pattern that way. So adding another check follows the pattern. But if this is a problem for you I can change it.